### PR TITLE
added more information to legs and fixed amount of created plans

### PR DESCRIPTION
**Pull request by carloscaldas**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/1976)

### CODE REVIEW

File: src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala [hunk 1]
```diff
@@ -176,50 +177,71 @@ class BeamScenarioLoader(
...
+    plansElements.groupBy(_.personId).foreach {
+      case (personId: PersonId, listOfElementsGroupedByPerson) =>
+        listOfElementsGroupedByPerson.groupBy(_.planIndex).foreach {
+          case (_, listOfElementsGroupedByPlan) if listOfElementsGroupedByPlan.nonEmpty =>
+            val person = population.getPersons.get(Id.createPersonId(personId.id))
+
+            val currentPlan = PopulationUtils.createPlan(person)
+            person.addPlan(currentPlan)
+
+            val personWithoutSelectedPlan = person.getSelectedPlan == null
```
Comment (REASY) [inline src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala]: Did not we agree to explicitly show that the plan is selected by a boolean flag when we write to CSV?

File: src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala [hunk 2]
```diff
@@ -176,50 +177,71 @@ class BeamScenarioLoader(
...
+    val leg = PopulationUtils.createAndAddLeg(currentPlan, planElement.legMode.getOrElse(""))
+    planElement.legDepartureTime.foreach(v => leg.setDepartureTime(v.toDouble))
+    planElement.legTravelTime.foreach(v => leg.setTravelTime(v.toDouble))
+
+    val legRoute = leg.getRoute
+    planElement.legRouteDistance.foreach(legRoute.setDistance)
+    planElement.legRouteStartLink.foreach(v => legRoute.setStartLinkId(Id.create(v, classOf[Link])))
+    planElement.legRouteEndLink.foreach(v => legRoute.setEndLinkId(Id.create(v, classOf[Link])))
+    planElement.legRouteTravelTime.foreach(v => legRoute.setTravelTime(v))
+    // TODO: could not find way to setup planElement.legRouteLinks and planElement.legRouteType
```
Comment (REASY) [inline src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala]: Are you about how to add the route to the leg? Please, look to the example https://github.com/LBNL-UCB-STI/beam/blob/develop/src/main/java/beam/physsim/jdeqsim/AgentSimToPhysSimPlanConverter.java#L423

File: src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala [hunk 3]
```diff
@@ -176,50 +177,71 @@ class BeamScenarioLoader(
...
+    act
+  }
+
+  private def buildAndAddLegToPlan(currentPlan: Plan, planElement: PlanElement): Leg = {
+    val leg = PopulationUtils.createAndAddLeg(currentPlan, planElement.legMode.getOrElse(""))
+    planElement.legDepartureTime.foreach(v => leg.setDepartureTime(v.toDouble))
+    planElement.legTravelTime.foreach(v => leg.setTravelTime(v.toDouble))
+
+    val legRoute = leg.getRoute
+    planElement.legRouteDistance.foreach(legRoute.setDistance)
```
Comment (REASY) [inline src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala]: Why I don't see the changes for `PlanElement` case class with those extra fields?

File: src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala (hunk 1 above)
Comment (REASY) [reply src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala]: Just have seen https://github.com/LBNL-UCB-STI/beam/pull/1974, it explains

File: src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala (hunk 3 above)
Comment (REASY) [reply src/main/scala/beam/utils/scenario/BeamScenarioLoader.scala]: I see those changes in https://github.com/LBNL-UCB-STI/beam/pull/1974

**Comment by JustinPihony** [comment]
test!

**Pull request merged**